File: standard-udfs-examples/standard-udfs-example-udfs-spark/src/test/scala/com/linkedin/stdudfs/examples/spark/TestArrayFillFunctionWrapper.scala
/**
 * Copyright 2018 LinkedIn Corporation. All rights reserved.
 * Licensed under the BSD-2 Clause license.
 * See LICENSE in the project root for license information.
 */
package com.linkedin.stdudfs.examples.spark

import com.linkedin.stdudfs.spark.common.AssertSparkExpression._
import org.apache.spark.sql.SparkSession
import org.testng.annotations.{BeforeClass, Test}

import scala.collection.mutable.WrappedArray._

class TestArrayFillFunctionWrapper {

  @BeforeClass
  def registerFunction(): Unit = {
    registerStandardUdf(
      "array_fill",
      classOf[ArrayFillFunctionWrapper]
    )
  }

  @Test
  def testArrayFill(): Unit = {
    assertFunction("array_fill(1, 5L)", make(Array(1, 1, 1, 1, 1)))
    assertFunction("array_fill('1', 5L)", make(Array("1", "1", "1", "1", "1")))
    assertFunction("array_fill(true, 5L)", make(Array(true, true, true, true, true)))
    assertFunction("array_fill(array(1), 5L)",
      make(Array(make(Array(1)), make(Array(1)), make(Array(1)), make(Array(1)), make(Array(1)))))
    assertFunction("array_fill(1, 0L)", make(Array()))
    assertFunction("array_fill(1, null)", null)
    assertFunction("array_fill(null, 1L)", null)
  }

  @Test
  def testArrayFillDataframe(): Unit = {
    val spark = SparkSession.builder().getOrCreate()
    import org.apache.spark.sql.functions._
    import spark.sqlContext.implicits._
    val df = 0.until(5).toDF()  // scalastyle:ignore magic.number
    df.select(callUDF("array_fill", col("value"), lit(5L))).collect()  // scalastyle:ignore magic.number
  }
}

File: transportable-udfs-examples/transportable-udfs-example-udfs-avro/src/test/java/com/linkedin/transport/examples/avro/TestArrayFillFunctionWrapper.java
/**
 * Copyright 2018 LinkedIn Corporation. All rights reserved.
 * Licensed under the BSD-2 Clause license.
 * See LICENSE in the project root for license information.
 */
package com.linkedin.transport.examples.avro;

import java.util.Arrays;
import org.apache.avro.Schema;
import org.testng.annotations.Test;

import static com.linkedin.transport.avro.common.AssertAvroUdf.*;
import static com.linkedin.transport.avro.common.SchemaFactory.*;


public class TestArrayFillFunctionWrapper {
  @Test
  public void testArrayFill() {
    assertFunction(new ArrayFillFunctionWrapper(), new Schema[]{INTEGER, LONG}, new Object[]{1, 5L},
        Arrays.asList(1, 1, 1, 1, 1));

    assertFunction(new ArrayFillFunctionWrapper(), new Schema[]{STRING, LONG}, new Object[]{"1", 5L},
        Arrays.asList("1", "1", "1", "1", "1"));

    assertFunction(new ArrayFillFunctionWrapper(), new Schema[]{array(INTEGER), LONG},
        new Object[]{Arrays.asList(1), 5L},
        Arrays.asList(Arrays.asList(1), Arrays.asList(1), Arrays.asList(1), Arrays.asList(1), Arrays.asList(1)));

    assertFunction(new ArrayFillFunctionWrapper(), new Schema[]{INTEGER, LONG}, new Object[]{1, 0L}, Arrays.asList());

    assertFunction(new ArrayFillFunctionWrapper(), new Schema[]{INTEGER, LONG}, new Object[]{1, null}, null);

    assertFunction(new ArrayFillFunctionWrapper(), new Schema[]{INTEGER, LONG}, new Object[]{null, 5}, null);
  }
}
